### PR TITLE
win,fs: use CreateDirectoryW instead of _wmkkdir

### PR DESCRIPTION
**Pull request by musm**
close https://github.com/libuv/libuv/issues/2499

### CODE REVIEW

File: src/win/fs.c [hunk 1]
```diff
@@ -1221,7 +1221,7 @@ void fs__unlink(uv_fs_t* req) {
 
 void fs__mkdir(uv_fs_t* req) {
   /* TODO: use req->mode. */
-  req->result = _wmkdir(req->file.pathw);
+  req->result = CreateDirectoryW(req->file.pathw,NULL);
```
Comment (musm) [inline src/win/fs.c:1224]: Does libuv do anything for the security descriptors?

File: src/win/fs.c [hunk 2]
```diff
@@ -1221,7 +1221,7 @@ void fs__unlink(uv_fs_t* req) {
...
   if (req->result == -1) {
     req->sys_errno_ = _doserrno;
     req->result = req->sys_errno_ == ERROR_INVALID_NAME
```
Comment (vtjnash) [inline src/win/fs.c:1227]:
```suggestion
    SET_REQ_WIN32_ERROR(req, GetLastError());
    if (req->sys_errno_ == ERROR_INVALID_NAME)
      req->result = UV_EINVAL;
```

**Comment by musm** [comment]
ty for the review

**Comment by vtjnash** [comment]
CI: https://ci.nodejs.org/job/libuv-test-commit/1968/

**Comment by vtjnash** [comment]
That does not seem to have gone especially well: https://ci.nodejs.org/job/libuv-test-commit-windows-cmake/188/testReport/

**Comment by musm** [comment]
I wonder if that was built on an older commit? I forced pushed the changes after committing your suggestions, almost at the same time you posted your CI comment, but realized the github web interface for committing the suggestions didn't apply cleanly and had to manually intervene and force push a fix.

**Comment by vtjnash** [comment]
Most of them probably built 4b9b394d1671fe5785f5237abfab03fcdd4f39f1 (it's listed as that in the logs for several of them)

**Comment by musm** [comment]
Strange, cross-compilation via Cygwin works and the tests pass. Not sure I want to download VS Code and all the deps to test it out via Cmake.

File: src/win/fs.c [hunk 3]
```diff
@@ -1221,12 +1221,11 @@ void fs__unlink(uv_fs_t* req) {
 
 void fs__mkdir(uv_fs_t* req) {
   /* TODO: use req->mode. */
-  req->result = _wmkdir(req->file.pathw);
+  req->result = CreateDirectoryW(req->file.pathw, NULL);
   if (req->result == -1) {
```
Comment (vtjnash) [inline src/win/fs.c:1225]:
```suggestion
  if (CreateDirectoryW(req->file.pathw, NULL)) {
    SET_REQ_RESULT(req, 0);
  } else {
```

**Comment by vtjnash** [comment]
CI: https://ci.nodejs.org/job/libuv-test-commit/1972/

File: src/win/fs.c (hunk 3 above)
Comment (musm) [reply src/win/fs.c]: great thank you, that seems to fix the cmake build

**Comment by vtjnash** [comment]
Ah, sorry, this broke the integration tests (https://ci.nodejs.org/job/node-test-binary-windows-js-suites/5162/RUN_SUBSET=3,nodes=win2012r2-COMPILED_BY-vs2019-x86/testReport/junit/(root)/test/parallel_test_fs_error_messages/), since I missed updating several more macros that access the implicit errno state. Fix incoming.

**Pull request merged**